hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/exception/QuotaExceededException.java

```diff
@@ -13,31 +13,75 @@
 import org.eclipse.hawkbit.repository.model.BaseEntity;
 
 /**
- * Thrown if too many entries are added to repository.
+ * Thrown if quota is exceeded (too many entries, file size, storage size).
  *
  */
 public final class QuotaExceededException extends AbstractServerRtException {
 
     private static final long serialVersionUID = 1L;
 
+    private static final String KB = "KB";
+    private static final String MB = "MB";
+
     private static final String ASSIGNMENT_QUOTA_EXCEEDED_MESSAGE = "Quota exceeded: Cannot assign %s more %s entities to %s '%s'. The maximum is %s.";
+    private static final String MAX_ARTIFACT_SIZE_EXCEEDED = "Maximum artifact size (%s) exceeded.";
+    private static final String MAX_ARTIFACT_SIZE_TOTAL_EXCEEDED = "Storage quota exceeded, %s left.";
+
+    private final QuotaType quotaType;
+
+    private final long exceededQuotaValue;
+
+    /**
+     * Enum that describes the types of quota that may be exceeded
+     */
+    public enum QuotaType {
+        STORAGE_QUOTA(SpServerError.SP_STORAGE_QUOTA_EXCEEDED, "message.upload.storageQuota"),
+        SIZE_QUOTA(SpServerError.SP_FILE_SIZE_QUOTA_EXCEEDED, "message.upload.fileSizeQuota"),
+        ASSIGNMENT_QUOTA(SpServerError.SP_QUOTA_EXCEEDED, "message.upload.artifactCountQuota");
```

Comment:
Additionally, I would not map the message keys, that are only relevant for the UI in Exception classes, but either directly in place where the exception is caught. There you will be able to use UIMessageIdProvider